pytact/data_reader.py

```diff
@@ -781,6 +788,16 @@ def is_file_representative(self) -> bool:
         Se also `Dataset.representative`."""
         return self._lreader[self._graph].representative == self.node.nodeid
 
+    @classmethod
+    def _group_by_clusters(cls, definitions : Iterable[Definition]) -> Iterable[list[Definition]]:
```

Comment:
```suggestion
    def _group_by_clusters(cls, definitions : Iterable[Definition]) -> Iterable[Iterable[Definition]]:
```
I think that with the new code, the type is no longer correct?

Reply:
to keep the original type, there is the `lambda x: list(x[1])`, I could keep it as an only iterator with replacing it by `lambda x: x[1]`

Reply:
Oh right, I missed that. I think an `Iterable[Iterable[Definition]]` is better. People can always cast it to a list.

Reply:
ok, I will change it, and test the code tomorrow (I expect some bugs, it is not tested so far)

Reply:
I changed it, tested it, and consequently changed it back :-) -- without the packing into list, itertools.groupby is fragile